src/PSRule.Rules.Azure/rules/Azure.VMSS.Rule.ps1

```diff
@@ -44,4 +44,23 @@ Rule 'Azure.VMSS.ComputerName' -Ref 'AZR-000262' -Type 'Microsoft.Compute/virtua
     Match 'Properties.virtualMachineProfile.osProfile.computerNamePrefix' $matchExpression
 }
 
+# Synopsis: Use SSH keys instead of common credentials to secure virtual machine scale sets against malicious activities.
+Rule 'Azure.VMSS.PublicKey' -Ref 'AZR-000288' -Type 'Microsoft.Compute/virtualMachineScaleSets' -If { IsLinuxVMSS } -Tag @{ release = 'GA'; ruleSet = '2022_09' } {
+    $Assert.In($TargetObject, 'properties.virtualMachineProfile.OsProfile.linuxConfiguration.disablePasswordAuthentication', $True).
+    Reason($LocalizedData.VMSSPublicKey, $TargetObject.name)
```

Comment:
Basically the same, but in some circumstances a different property for the name might be used such as `resourceName` instead of `name`.

```suggestion
    Reason($LocalizedData.VMSSPublicKey, $PSRule.TargetName)
```